app/models/room.rb

```diff
@@ -9,13 +9,15 @@ class Room < ApplicationRecord
   def self.search(search)
     # If the argument search not nil, then a block is run where the track object is found based on params[:search]
     if search.present?
+      search = search.downcase
       begin
-        track = Track.find_by(artist: search.downcase).rooms.first
+        track = Track.where('artist LIKE ?', "%#{search}%").map(&:rooms).first
       rescue StandardError
         nil
       end
       # then the rooms with the searched track are all returned through the room_track
-      Track.where(artist: search.downcase).map(&:rooms).flatten if track
+      room_names_arr = Track.where('artist LIKE ?', "%#{search}%").map(&:rooms).flatten.pluck(:name).uniq
+      Room.where(name: [room_names_arr]) if track
     else
       ''
     end
```

Comment:
### Note 1

1. We can use guard clauses to minimize nested conditionals
2. Logic of line 14 is the same with 19. Both returns the rooms with the search tracked.
3. We can use the `bang` operator to downcase the search term 😃 

```ruby
def self.search(search)
  # If the argument search not nil, then a block is run where the track object is found based on params[:search]
  return if search.nil?

  search.downcase!
  # then the rooms with the searched track are all returned through the room_track
  Track.where('artist LIKE ?', "%#{search}%").map(&:rooms).flatten.uniq
rescue StandardError
  nil
end
```

Note that the code below will have an impact on the `rooms_controller.rb`. As follows:

1. If there are no results based on the query, the `where` method returns `[]`, which is `true`. Also, `[]` is not considered as `nil`. So, we need to check using the `empty?` or `any?` method 
2. Added `params[:search].present?` in the `elsif` block so that the search method will be used only when there is a search parameter
Before
```ruby
def index
  if params[:search].present? && Room.search(params[:search])
    ...
  elsif Room.search(params[:search]).nil?
    ...
  else
    ...
  end
end
```
After
```ruby
def index
  if params[:search].present? && Room.search(params[:search]).any?
    ...
  elsif params[:search].present? && Room.search(params[:search]).empty?
    ...
  else
    ...
  end
end
```